File: C++/Player.cpp
//Honor Pledge: I pledge that I have neither given nor received any help on this assignment
#include <iostream>
#include "Player.h"
#include <string.h>
#include <ctime>
#include <cstdlib>
char GAME[] = {"HORSE"};
player::player() {
	if(!this->init){
	this->init=true;
	srand(time(0));
	}
	strcpy(this->horse, " ");
	
}

int player::Randshot() {
	int N;
	N = rand() % 2 + 1;
	return N;
}

char player::sethorse(int n) {
	if (n == 0) {
		strcpy(this->horse, &GAME[n]);
	}
	else {
		strcat(this->horse, &GAME[n]);
	}
	return this->horse[n];
}
void player::gethorse() {
	int p=0;
	while (p < 5) {
		std::cout << this->horse[p];
		p++;
	}
	std::cout << "\n";
}
player::~player(){

}

File: Java/Player.java
//Honor Pledge: I pledge that I have neither given nor received any help on this assignment
import java.io.*;
import java.util.*;


public class Player
{
	private char[] horse; 
	private int shot;
	private Random rand = new Random();
	public static char[] WORD = {'H', 'O', 'R', 'S', 'E'};
	
	public Player()
	{
		this.horse = new char[5];
	}
	public int getshot(){
		this.shot= rand.nextInt(2)+1;
		return this.shot;
	}
	void gethorse(){
		int i=0;
		while(i < 5){
			System.out.print(this.horse[i]);
			i++;
		}
	}
	char sethorse(int n){
		this.horse[n] = WORD[n];
		return this.horse[n];
	}
}